test/integration/client/client_test.go

```diff
@@ -1342,3 +1350,112 @@ func TestExtractModifyApply_ForceOwnership(t *testing.T) {
 		t.Errorf("createMgrExtracted apply configuration did not match expected, got:\n%s\n", cmp.Diff(expectedCreateExtracted, createMgrExtracted))
 	}
 }
+
+// regression test for https://issues.k8s.io/117258
+func TestAPIServerTransportMetrics(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllAlpha", true)()
+	defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllBeta", true)()
+
+	result := kubeapiservertesting.StartTestServerOrDie(t, nil, []string{"--disable-admission-plugins", "ServiceAccount"}, framework.SharedEtcd())
+	defer result.TearDownFn()
+
+	client := clientset.NewForConfigOrDie(result.ClientConfig)
+
+	hits1, misses1, entries1 := checkTransportMetrics(t, client)
+	// hit ratio at startup depends on multiple factors
+	// be conservative and expect only a 75 percent
+	if (hits1*100)/(hits1+misses1) < 75 {
+		t.Fatalf("transport cache hit ratio %d lower than 75 percent", (hits1*100)/(hits1+misses1))
+	}
+
+	aggregatorClient := aggregatorclient.NewForConfigOrDie(result.ClientConfig)
+	aggregatedAPI := &apiregistrationv1.APIService{
+		ObjectMeta: metav1.ObjectMeta{Name: "v1alpha1.wardle.example.com"},
+		Spec: apiregistrationv1.APIServiceSpec{
+			Service: &apiregistrationv1.ServiceReference{
+				Namespace: "kube-wardle",
+				Name:      "api",
+			},
+			Group:                "wardle.example.com",
+			Version:              "v1alpha1",
+			GroupPriorityMinimum: 200,
+			VersionPriority:      200,
+		},
+	}
+	_, err := aggregatorClient.ApiregistrationV1().APIServices().Create(context.Background(), aggregatedAPI, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < 100; i++ {
+		apiService, err := aggregatorClient.ApiregistrationV1().APIServices().Get(context.Background(), "v1alpha1.wardle.example.com", metav1.GetOptions{})
+		if err != nil {
+			t.Fatal(err)
+		}
+		// mutate the object
+		apiService.Labels = map[string]string{"key": fmt.Sprintf("val%d", i)}
+		_, err = aggregatorClient.ApiregistrationV1().APIServices().Update(context.Background(), apiService, metav1.UpdateOptions{})
+		if err != nil && !apierrors.IsConflict(err) {
```

Comment:
are we concerned at all about timeouts or network flakes in the series of 100 update requests?

Reply:
that will be symptom that something else wrong ... also, we need to have some "large numbers" to get closer to the average, if this fails and we swallow it we may have false positives or negatives

Reply:
did we run this repeatedly in CI to make sure this doesn't flake?

Reply:
reduced the number of requests and adding a check to avoid false positives
 run with stress
```
4m55s: 1831 runs so far, 0 failures
5m0s: 1868 runs so far, 0 failures
5m5s: 1898 runs so far, 0 failures
5m10s: 1928 runs so far, 0 failures
5m15s: 1965 runs so far, 0 failures
5m20s: 1995 runs so far, 0 failures
```